fast_llm/ext_utils/hf_auth.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
```

Comment:
How big of an addition would a hf_hub dependency be? We already have most of hf tools as dependencies anyway, and this shouldn't add much sub-dependencies...